newIDE/app/src/Export/LocalExporters/LocalFileSystem.spec.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import LocalFileSystem from './LocalFileSystem';
+import path from 'path';
+
+describe('LocalFileSystem', () => {
+  describe('file content storing and reading', () => {
+    test('it can mark files to be copied from an URL as to be downloaded', () => {
+      const localFileSystem = new LocalFileSystem({
+        downloadUrlsToLocalFiles: true,
+      });
+
+      localFileSystem.copyFile(
+        'http://file.com/from/url',
+        '/folder/downloaded-file'
+      );
+      expect(localFileSystem.getAllUrlFilesIn('/')).toEqual([
+        {
+          filePath: '/folder/downloaded-file',
+          url: 'http://file.com/from/url',
+        },
+      ]);
+      expect(localFileSystem.getAllUrlFilesIn('/folder/')).toEqual([
+        {
+          filePath: '/folder/downloaded-file',
```

Comment:
filepath stays the same and does not become relative?

Reply:
Yep they stay absolute. Not a problem because 1) the information is equivalent for the caller (get a relative or absolute path, it's the same as long as you know the base path if it's relative) and 2) it's actually easier to have absolute paths for the caller (it will be used to download files locally, so getting the absolute path is fine because it's already exactly where we want the download to be stored).

Reply:
(also, this is a fairly not generic implementation, just something that works well enough for exports - not something we expect to use outside)